### ISSUE TITLE
BatchImage.IsUseless() and checking white list

### ISSUE THREAD

**Issue by cukie** [cukie/SMIA#1]
the current method of sorting our white list entries and then checking against sorted entries introduces extra output sometimes...

E.g.
        "snaka under NOTmab11, stroma, NOTepi",
        "bk under snaka, stroma, NOTepi"

will also match
       NOTmab11 under snaka, stroma, NOTepi 

This is only an issue when markers become masks that are already valid in prior white list entries.

**Comment by cukie** [comment]
Important to note that the extra output generated is VALID output. It is just unwanted and could lead to confusion.

**Comment by cukie** [comment]
Another issue here is that this is terribly inefficient. Instead of generating all possibilities (which is O(n!) ), we should be parsing the input and generating only the Masks and Markers needed. That will resolve this bug as well as improve run times dramatically. The code in question is:

```
def MasksFromList(self, mask_list, combination_max):
        """
        Given a list of Mask or Marker objects
        returns a list of all combinations of masks
        in the form of a tuple. (mask_loc,name)
        """
        mask_tuples = []
        count = 0
        for lim in xrange(1, combination_max+1):
                for item in itertools.combinations(mask_list, lim):

                    # Create our list of names
                    names = [x.name for x in item]
                    # If this combination is not useless...
                    if not self.IsUseless(names):
                        count += 1
                        # TODO: A BatchImage object should really have a sense of what batch its operating on.
                        # Some meta data like the path of the batch should be sufficient to log meaningful info.
                        logger.debug("Processing Mask {0}".format(count))
                        # our original mask will have all indices in it
                        output_mask = np.arange(0, self.num_pixels)
                        for mask in item:
                            # like a cumsum of intersections...
                            output_mask = GetIntersection(output_mask, mask.masked_indices.flatten())
                        names = ', '.join(names)
                        mask_tuples.append((output_mask, names))

        return mask_tuples
```

**Comment by cukie** [comment]
Solved by PR: #18
